### Problem 1: Extending the Lady Tasting Tea Experiment

In this task, we extend the classic Lady Tasting Tea experiment by increasing the number of cups  from 8 (4 tea-first, 4 milk-first) to 12 (8 tea-first, 4 milk-first). We will estimate the probability of correctly identifying all cups by chance using NumPy simulations and compare the results with the original 8-cup experiment to assess extreme outcomes and discuss possible adjustments to the p-value threshold.



### Problem Breakdown:
 This problem is split into five parts for a step-by-step approach:
   1. Libraries: Import the necessary libraries.
   2. Part A: Simulate the Original Tea Experiment (8 cups of tea).
   3. Part B: Simulate the Extended Tea Experiment (12 cups of tea).
   4. Part C: Compare the Original Tea Experiment vs. Extended Tea Experiment.
   5. Conclusion: Interpret the results and discuss the implications for statistical significance.


### - Libraries:


In [87]:
import numpy as np  # For numerical operations and simulations

import math  # For mathematical functions

import matplotlib.pyplot as plt  # For visualizing results

import random  # For random selections

### Part A: Original Tea Experiment (8 Cups of Tea)
The original Lady Tasting Tea experiment involved 8 cups, 4 tea-first and 4 milk-first. We will calculate the probability of correctly identifying the tea-first cups by chance.


- Part A: Original Tea Experiment (8 Cups of Tea). The original Lady Tasting Tea experiment involved 8 cups, 4 tea-first and 4 milk-first. We will calculate the probability of correctly identifying the tea-first cups by chance.


In [88]:
# Number of cups in the original experiment:
no_cups = 8
no_cups_milk_first = 4
no_cups_tea_first = 4

In [89]:
# Calculate the number of combinations (order does not matter, no element is selected more than once):
comb = math.comb(no_cups, no_cups_milk_first)
print("The number of possible combinations is:", comb)

The number of possible combinations is: 70


In [90]:
# Calculate the probability of selecting the correct cups
prob = 1 / comb
print('Probability of correctly identifying the Original Tea Experiment (8 Cups of Tea) is:', prob)


Probability of correctly identifying the Original Tea Experiment (8 Cups of Tea) is: 0.014285714285714285


In [91]:
# Simulation using NumPy to verify the theoretical probability
# Set up the array representing the cups (1 = tea-first, 0 = milk-first)
cups_8 = np.array([1]*4 + [0]*4)

# Number of simulations
trials = 150000
successes = 0

# Simulation loop
for _ in range(trials):
    np.random.shuffle(cups_8)  # Shuffle the cups randomly
    guessed_cups = cups_8[np.random.choice(range(8), size=4, replace=False)]  # Randomly select 4 cups
    
    if np.sum(guessed_cups) == 4:  # Check if all selected cups are tea-first
        successes += 1

# Estimate probability from the simulation
simulated_probability = successes / trials
print("Estimated probability from simulation:", simulated_probability)


Estimated probability from simulation: 0.01398


### Part B: Extended Tea Experiment (12 Cups of Tea)
- In this part, we extend the experiment by increasing the total number of cups to 12, with 8 tea-first cups and 4 milk-first cups. We will calculate the new probability and compare it with the original experiment.

In [92]:
# Number of cups in the extended experiment
no_cups_ext = 12
no_cups_milk_first_ext = 4
no_cups_tea_first_ext = 8

In [93]:
# Calculate the number of combinations for the extended experiment
comb_ext = math.comb(no_cups_ext, no_cups_milk_first_ext)
print("The number of possible combinations is:", comb_ext)


The number of possible combinations is: 495


In [94]:
# Calculate the probability of correctly identifying the cups
prob_ext = 1 / comb_ext
print('Probability of correctly identifying the Extended Tea Experiment (12 Cups of Tea) is:', prob_ext)

Probability of correctly identifying the Extended Tea Experiment (12 Cups of Tea) is: 0.00202020202020202


In [95]:

# Simulation using NumPy to verify the extended experiment
# Set up the array representing the cups in the extended experiment
cups_12 = np.array([1]*8 + [0]*4)

# Number of simulations
successes_12 = 0

# Simulation loop for the extended experiment
for _ in range(trials):
    np.random.shuffle(cups_12)  # Shuffle the cups randomly
    guessed_cups = cups_12[np.random.choice(range(12), size=8, replace=False)]  # Select 8 cups
    
    if np.sum(guessed_cups) == 8:  # Check if all selected cups are tea-first
        successes_12 += 1

# Estimate probability for the extended experiment
simulated_probability_12 = successes_12 / trials
print("Estimated probability from simulation (12 cups):", simulated_probability_12)

Estimated probability from simulation (12 cups): 0.0019866666666666665


- Part C: Comparing Original Tea Experiment vs. Extended Tea Experiment. The extended 12-cup experiment makes it significantly harder to guess all cups correctly by chance. The probability drops from 1 in 70 (Original Tea Experiment) to 1 in 495 (Extended Tea Experiment). This is more than a 7-fold decrease in chance success.

### Conclusion:

- The comparison of the original and extended experiments shows a significant difference in the likelihood of randomly guessing the cups correctly.
- The extended experiment (12 cups) is much more difficult to get correct by chance, making the evidence against random guessing stronger.
- The simulations confirm the theoretical calculations and demonstrate that the extended experiment is a more reliable test.